codegen-server/src/main/kotlin/software/amazon/smithy/rust/codegen/server/smithy/generators/protocol/ServerProtocolTestGenerator.kt

```diff
@@ -178,22 +191,26 @@ class ServerProtocolTestGenerator(
         writeInline("let expected =")
         instantiator.render(this, inputShape, httpRequestTestCase.params)
         write(";")
-        with(httpRequestTestCase) {
-            host.orNull()?.also { host ->
-                val withScheme = "http://$host"
-                rustTemplate(
-                    """
-                    let mut http_request = http_request;
-                    let ep = #{SmithyHttp}::endpoint::Endpoint::mutable(#{Http}::Uri::from_static(${withScheme.dq()}));
-                    ep.set_endpoint(http_request.uri_mut(), parts.acquire().get());
-                    """,
-                    *codegenScope,
-                )
-            }
-            resolvedHost.orNull()?.also { host ->
-                rust("""assert_eq!(http_request.uri().host().expect("host should be set"), ${host.dq()});""")
+        httpRequestTestCase.body.orNull()?.also { body ->
+            rustTemplate(
+                """
+                ##[allow(unused_mut)] let mut http_request = http::Request::builder()
+                    .uri(${httpRequestTestCase.uri.dq()})
+                    .body(#{SmithyHttpServer}::Body::from(#{Bytes}::from_static(b${body.dq()}))).unwrap();
+                """,
+                *codegenScope
+            )
+            if (!httpRequestTestCase.bodyMediaType.isEmpty()) {
+                rust("""http_request.headers_mut().insert("Content-Type", http::header::HeaderValue::from_static(${httpRequestTestCase.bodyMediaType.get().dq()}));""")
```

Comment:
I am not overly happy about this part of the tests generation. However I don't think it worth it to refactor it until we do not implement the rest of the protocol.